static/js/publisher/metrics/graphs/activeDevices.js

```diff
@@ -1,201 +1,223 @@
+import { format as d3Format } from 'd3-format';
+import { utcParse as d3TimeParse, utcFormat as d3TimeFormat } from 'd3-time-format';
+import { scaleLinear as d3ScaleLinear, scaleOrdinal as d3ScaleOrdinal } from 'd3-scale';
+import { schemePaired as d3SchemePaired } from 'd3-scale-chromatic';
+import { area as d3Area, stack as d3Stack, curveMonotoneX as d3CurveMonotoneX, stackOrderReverse as d3StackOrderReverse } from 'd3-shape';
+import { extent as d3Extent, bisector as d3Bisector } from 'd3-array';
+import { axisBottom as d3AxisBottom, axisLeft as d3AxisLeft } from 'd3-axis';
+import { cullXAxis, cullYAxis } from './helpers';
 
 function showGraph(el) {
-  formatAxis(el);
   el.style.opacity = 1;
 }
 
-export default function activeDevices(days, activeDevices, type) {
-  const el = document.getElementById('active_devices');
+export default function activeDevices(activeDevices) {
+  // Basic svg setup
+  const holder = document.querySelector('svg').parentNode;
+  const svg = d3Select('svg');
+  svg.attr('width', holder.clientWidth);
+  const margin = { top: 20, right: 0, bottom: 30, left: 50 };
+  const padding = { top: 0, right: 0, bottom: 16, left: 16 };
+  const width = svg.attr('width') - margin.left - margin.right - padding.left - padding.right;
+  const height = svg.attr('height') - margin.top - margin.bottom - padding.top - padding.bottom;
+  const g = svg.append('g').attr('transform', `translate(${margin.left},${margin.top})`);
+
+  // Prepare the data
+  const parseTime = d3TimeParse('%Y-%m-%d');
+  const _data = [];
+  const _keys = [];
+  let max = 0;
+  activeDevices.buckets.forEach((bucket, i) => {
+    let obj = {
+      date: parseTime(bucket)
+    };
+
+    activeDevices.series.forEach(series => {
+      obj[series.name] = series.values[i];
+      if (_keys.indexOf(series.name) < 0) {
+        _keys.push(series.name);
+      }
+    });
+
+    _data.push(obj);
+  });
+
+  _data.forEach(date => {
+    const _max = Object.keys(date).filter(key => key !== 'date').reduce((previous, key) => {
+      return previous + date[key];
+    }, 0);
 
-  // If there's no data, we still want to show the axis
-  let minYAxisValue;
-  if (days.length === 1) { // Only includes the 'x'
-    let offset = 0;
-    const start = moment.utc().startOf('day').subtract(30, 'days');
-    let dummyActiveDevices = [''];
+    if (_max > max) {
+      max = _max;
+    }
+  });
 
-    while(offset < 30) {
-      days.push(start.clone().add(offset, 'days'));
-      dummyActiveDevices.push(0);
-      offset++;
+  const shortValue = number => {
+    if (number < 1000) {
+      return number;
     }
-    activeDevices.push(dummyActiveDevices);
-    minYAxisValue = 1;
+    return d3Format('.2s')(number);
+  };
+
+  const commaValue = number => {
+    return d3Format(',')(number);
+  };
+
+  // Prepare the axis
+  const x = d3ScaleLinear().
+    rangeRound([padding.left, width - padding.left - padding.right]);
```

Comment:
That's an odd way to do chained functions. Without . in the beginning it looks like independent function call that was wrongly indented. I guess more common way is 

```
something()
  .somethingElse()
  .somethingCompletelyDifferent()
```

Reply:
That's what I thought, but for some reason my IDE complained!? I think I need to check the settings